feature-libs/cart/base/occ/adapters/occ-cart-guest-user.adapter.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Injectable, inject } from '@angular/core';
+import {
+  CART_GUEST_USER_NORMALIZER,
+  CartGuestUserAdapter,
+} from '@spartacus/cart/base/core';
+import { CartGuestUser } from '@spartacus/cart/base/root';
+import {
+  ConverterService,
+  InterceptorUtil,
+  LoggerService,
+  OccEndpointsService,
+  USE_CLIENT_TOKEN,
+  backOff,
+  isServerError,
+  tryNormalizeHttpError,
+} from '@spartacus/core';
+import { Observable, catchError } from 'rxjs';
+
+const CONTENT_TYPE_JSON_HEADER = { 'Content-Type': 'application/json' };
+
+@Injectable()
+export class OccCartGuestUserAdapter implements CartGuestUserAdapter {
+  protected logger = inject(LoggerService);
+  protected http = inject(HttpClient);
+  protected occEndpointsService = inject(OccEndpointsService);
+  protected converterService = inject(ConverterService);
+
+  protected getRequestHeaders(): HttpHeaders {
+    return InterceptorUtil.createHeader(
+      USE_CLIENT_TOKEN,
+      true,
+      new HttpHeaders({
+        ...CONTENT_TYPE_JSON_HEADER,
+      })
+    );
+  }
+
+  createCartGuestUser(
+    userId: string,
+    cartId: string,
+    guestUserDetails?: CartGuestUser
```

Comment:
[MEDIUM]
Allowing here for the full CartGuestUser looks misleading to me.
In the swagger docs I can see the only allowed property is `email`, but `uid` and `name` cannot be passed in the payload in the request to OCC.
see
https://api.cp96avkh5f-integrati2-s1-public.model-t.cc.commerce.ondemand.com/occ/v2/swagger-ui/index.html#/Carts/createCartGuestUser
and
https://api.cp96avkh5f-integrati2-s1-public.model-t.cc.commerce.ondemand.com/occ/v2/swagger-ui/index.html#/Carts/updateCurrentUserProfile

Reply:
Already answered offline my Mateusz: in the future other properties might be also possible to pass for the backend.